Makefile.am

```diff
@@ -224,6 +224,23 @@ clean-local::
 maintainer-clean-local::
 	rm -rf dist/ node_modules/ package-lock.json
 
+BUILT_SOURCES += package-lock.json
+
+package-lock.json: package.json
+	@echo ''
+	@echo '    package-lock.json needs to be (re)generated'
+	@echo ''
+	@echo '    Please run:'
+	@echo ''
+	@echo '        make npm'
+	@echo ''
+	@false
```

Comment:
As discussed earlier, I think that strikes the right balance in preventing weird bugs and not accidentally hitting the network on `make check`.